static/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.tsx

```diff
@@ -277,6 +282,90 @@ export default class SentryApplicationDetails extends AsyncView<Props, State> {
     }
   };
 
+  getAvatarPreview = (avatarStyle: 'color' | 'simple') => {
+    const {app} = this.state;
+    const styleMap = {
+      color: {
+        size: 50,
+        title: t('Default Logo'),
+        description: t('The default icon for integrations'),
+      },
+      simple: {
+        size: 20,
+        title: t('Default Icon'),
+        description: t('This is an optional icon used for Issue Linking'),
+      },
+    };
+    return (
+      app && (
+        <AvatarPreview>
+          <StyledPreviewAvatar
+            size={styleMap[avatarStyle].size}
+            sentryApp={app}
+            isDefault
+          />
+          <AvatarPreviewTitle>{styleMap[avatarStyle].title}</AvatarPreviewTitle>
+          <AvatarPreviewText>{styleMap[avatarStyle].description}</AvatarPreviewText>
+        </AvatarPreview>
+      )
+    );
+  };
+
+  getAvatarModel = (avatarStyle: 'color' | 'simple'): Model => {
+    const {app} = this.state;
+    const defaultModel: Model = {
+      avatar: {
+        avatarType: 'default',
+        avatarUuid: null,
+      },
+    };
+    if (!app) {
+      return defaultModel;
+    }
+    const isColor = avatarStyle === 'color';
+    return {
+      avatar:
+        (app?.avatars || []).find(({color}) => color === isColor) || defaultModel.avatar,
+    };
+  };
+
+  getAvatarChooser = (avatarStyle: 'color' | 'simple') => {
+    const {app} = this.state;
+    if (!app) {
+      return null;
+    }
+    const isColor = avatarStyle === 'color';
+
+    return (
+      <Feature features={['organizations:sentry-app-logo-upload']}>
+        <AvatarChooser
+          type={isColor ? 'sentryAppColor' : 'sentryAppSimple'}
+          allowGravatar={false}
+          allowLetter={false}
+          endpoint={`/sentry-apps/${app.slug}/avatar/`}
+          model={this.getAvatarModel(avatarStyle)}
+          onSave={({avatar}) => {
```

Comment:
I'd do something like:
```tsx
addAvatar = model => {
  const {avatar} = model;
  const {app} = this.state;
  if (avatar) {
    const prevAvatars = app?.avatars ?? [];
    const avatars = prevAvatars.filter(prevAvatar => prevAvatar.color !== avatar.color);
    avatars.push(avatar);
    this.setState({app: {...app, avatars}});
  }
};
```
so that on this line you could do:
```tsx
onSave={this.addAvatar}
```

Reply:
done!